miner/scroll_worker.go

```diff
@@ -650,14 +652,30 @@ func (w *worker) handlePipelineResult(res *pipeline.Result) error {
 	}
 
 	var commitError error
-	if res != nil && res.FinalBlock != nil {
+	if res.FinalBlock != nil {
 		if commitError = w.commit(res); commitError == nil {
 			return nil
 		}
 		log.Error("Commit failed", "header", res.FinalBlock.Header, "reason", commitError)
+		if _, isRetryable := commitError.(retryableCommitError); !isRetryable {
```

Comment:
Could you please explain how this solved the `panic: runtime error: invalid memory address or nil pointer dereference` issue?

Reply:
This PR was originally meant to fix non-authorized signer with --mine flag enabled. Which required me to remove lines L766-767 from `commit()`, which were the lines that triggered the nil pointer panic.